command/agent/consul/group_test.go

```diff
@@ -67,33 +77,43 @@ func TestConsul_Connect(t *testing.T) {
 		return len(services) == 2
 	}, 3*time.Second, 100*time.Millisecond)
 
-	services, err := consulClient.Agent().Services()
-	require.NoError(t, err)
-	require.Len(t, services, 2)
+	// Test a few times to ensure Nomad doesn't improperly deregister
+	// Connect services.
+	for i := 10; i > 0; i-- {
+		services, err := consulClient.Agent().Services()
+		require.NoError(t, err)
+		require.Len(t, services, 2)
 
-	serviceID := MakeTaskServiceID(alloc.ID, "group-"+alloc.TaskGroup, tg.Services[0], false)
-	connectID := serviceID + "-sidecar-proxy"
+		serviceID := MakeTaskServiceID(alloc.ID, "group-"+alloc.TaskGroup, tg.Services[0], false)
+		connectID := serviceID + "-sidecar-proxy"
 
-	require.Contains(t, services, serviceID)
-	agentService := services[serviceID]
-	require.Equal(t, agentService.Service, "testconnect")
-	require.Equal(t, agentService.Address, "10.0.0.1")
-	require.Equal(t, agentService.Port, 9999)
-	require.Nil(t, agentService.Connect)
-	require.Nil(t, agentService.Proxy)
+		require.Contains(t, services, serviceID)
+		//require.True(t, isNomadService(serviceID))
+		//require.False(t, isNomadSidecar(serviceID))
```

Comment:
Are these assertions left over from debugging or should they be uncommented?

Reply:
Ha, I commented them out to run the test against master and forgot to uncomment them. Thanks!

(The good news is the test reliably fails against master, so I have high confidence in the fix.)